src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs

```diff
@@ -324,10 +325,33 @@ private static void OnItemsRepeaterElementPrepared(ItemsRepeater sender, Microso
 
 		SetItemSelection(sender, args.Element, selected);
 	}
+
 	private static void OnItemsRepeaterItemsSourceChanged(DependencyObject sender, DependencyProperty dp)
 	{
 		// When we reach here, ItemsSourceView is already updated.
 		if (sender is not ItemsRepeater ir) return;
+
+		SynchronizeDefaultSelection(ir);
+
+#if HAS_UNO_WINUI || __WASM__
+		ir.ItemsSourceView.CollectionChanged += OnItemsSourceViewCollectionChanged;
+
+		var collectionChangedDisposable = Disposable.Create(() => ir.ItemsSourceView.CollectionChanged -= OnItemsSourceViewCollectionChanged);
+
+		if (GetSelectionSubscription(ir) is CompositeDisposable cd)
+		{
+			cd.Add(collectionChangedDisposable);
+		}
```

Comment:
> you should:
> 
> * create a seperate SerialDisposable for this subscription
> * add that serial-disp to the composite-disp
> * and clear then set (dont just re-assign) the inner disposable here: (_SourceViewCollectionChangedDisposable as SerialDisposable).Disposable = Disposable.Create(() => ir.ItemsSourceView.CollectionChanged -= OnItemsSourceViewCollectionChanged)

@Xiaoy312 does this work or do we need to set the SelectionSubscription property directly?

Reply:
SelectionSubscription? we dont touch that here
and what i mean was `_SourceViewCollectionChangedDisposable` was an attached-property to disposable not a local field

Reply:
essentially, you would have two disposables:
1. one for everything (which comprise of the one below)
2. one just for ItemsSource/View

also, make sure that the SelectionSubscription only contains a SINGLE instance of the items-source disposable
and you are only setting/clearing the serial-disposable.disposable part of the items-source disposable
lmk if that wasnt clear enough for you, i can push the change directly

Reply:
The attached-property makes sense but i'm having trouble seeing how we can add the ItemsSource/View disposable to the everything disposable because of

> the crash is normal, since ItemsSourceView is updated only when ItemsSource is likely at the time of this extensions being set, that could be null still

Do we need to move this? ```ir.RegisterDisposablePropertyChangedCallback(ItemsRepeater.ItemsSourceProperty, OnItemsRepeaterItemsSourceChanged)```
I am probably not following well, if you could push the changes i can see what I misunderstood

Reply:
see my last commit